core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -62,7 +64,23 @@ class UnionRDD[T: ClassTag](
     var rdds: Seq[RDD[T]])
   extends RDD[T](sc, Nil) {  // Nil since we implement getDependencies
 
+  // Evaluate partitions in parallel. Partitions of each rdd will be cached by the `partitions`
+  // val in `RDD`.
+  private[spark] lazy val parallelPartitionEval: Boolean = {
```

Comment:
This took me a minute to understand. It's a boolean val but really that's a dummy. This is evaluated only for side effects. It's only evaluated one place the partitions are used and not the other. If it's only really helpful in `getPartitions`, how about just making the `rdds.zipWithIndex` parallel and let it process that way? no dummy lazy vals.

I also think a custom parallelism and fork join pool are overkill. Just use the default pool. I'd like to avoid yet more knobs to tune unless it's strongly needed.

Is there any downside to parallel execution?